docs/content/AutoGPT/component agent/protocols.md

```diff
@@ -0,0 +1,165 @@
+# ⚙️ Protocols
+
+Protocols are *interfaces* implemented by [Components](./components.md) used to group related functionality. Each protocol needs to be handled explicitly by the agent at some point of the execution. We provide a comprehensive list of built-in protocols that are already handled in the built-in `Agent`, so when you inherit from the base agent all built-in protocols will work!
+
+**Protocols are listed in the order of the default execution.**
+
+## Order-independent protocols
+
+Components implementing exclusively order-independent protocols can added in any order, including in-between ordered protocols.
+
+### `DirectiveProvider`
+
+Yields constraints, resources and best practices for the agent. This is purely informational and will be passed to a llm after prompt is ready using `BuildPrompt` protocol.
+
```

Comment:
Do I need this if I write a command to have my command listed?

Reply:
No, why this idea? And the prompt building was reverted back to the old system for now, there's no `BuildPrompt` protocol:
> ... This has no direct impact on other protocols; is purely informational and will be passed to a llm when the prompt is built.

Reply:
Put that in the docs